bin/hog/batchsend.C

```diff
@@ -93,33 +72,21 @@ std::ostream& operator<<(std::ostream& o, const std::vector<Destination>& xs) {
   return o;
 }
 
-void sendFileContents(int connection, const openfd& sfd) {
-  struct stat sb;
-  fstat(sfd.fd(), &sb);
-
-  // let the remote side know how big this file is
-  size_t fsize = sb.st_size;
-  ssend(connection, (const uint8_t*)&fsize, sizeof(fsize));
-
+void sendFileContents(NetConnection& connection, const openfd& sfd) {
   // send the entire file contents
-  off_t offset = 0;
-  while (offset < sb.st_size) {
-    if (sendfile(connection, sfd.fd(), &offset, sb.st_size) == -1) {
-      if (errno != EAGAIN) {
-        throw std::runtime_error("error trying to send file: " + std::string(strerror(errno)));
-      }
-    }
+  if (! connection.sendFile(sfd.fd())) {
+    throw std::runtime_error(strerror(errno));
```

Comment:
should `sendFile()` failed, could we find additional information to locate the source of exception from error message?

Reply:
You mean,

    throw std::runtime_error("sendFile failed: " + std::string(strerror(errno)));

right?